worlds/aquaria/Regions.py

```diff
@@ -0,0 +1,1377 @@
+    return (state.has("Energy form", player) or
+            state.has("Nature form", player) or
+            state.has("Beast form", player) or
+            state.has("Li and Li song", player) or
+            state.has("Baby nautilus", player) or
+            state.has("Baby piranha", player) or
+            state.has("Baby blaster", player)
+            )
+
+
+def _has_shield_song(state, player: int) -> bool:
+    """`player` in `state` has the shield song item"""
+    return state.has("Shield song", player)
+
+
+def _has_bind_song(state, player: int) -> bool:
+    """`player` in `state` has the bind song item"""
+    return state.has("Bind song", player)
+
+
+def _has_energy_form(state, player: int) -> bool:
+    """`player` in `state` has the energy form item"""
+    return state.has("Energy form", player)
+
+
+def _has_beast_form(state, player: int) -> bool:
+    """`player` in `state` has the beast form item"""
+    return state.has("Beast form", player)
+
+
+def _has_nature_form(state, player: int) -> bool:
+    """`player` in `state` has the nature form item"""
+    return state.has("Nature form", player)
+
+
+def _has_sun_form(state, player: int) -> bool:
+    """`player` in `state` has the sun form item"""
+    return state.has("Sun form", player)
+
+
+def _has_light(state, player: int) -> bool:
+    """`player` in `state` has the light item"""
+    return state.has("Baby dumbo", player) or _has_sun_form(state, player)
+
+
+def _has_dual_form(state, player: int) -> bool:
+    """`player` in `state` has the dual form item"""
+    return _has_li(state, player) and state.has("Dual form", player)
+
+
+def _has_fish_form(state, player: int) -> bool:
+    """`player` in `state` has the fish form item"""
+    return state.has("Fish form", player)
+
+
+def _has_spirit_form(state, player: int) -> bool:
+    """`player` in `state` has the spirit form item"""
+    return state.has("Spirit form", player)
+
+
+def _has_big_bosses(state, player: int) -> bool:
+    """`player` in `state` has beated every big bosses"""
+    return (state.has("Fallen God beated", player) and
+            state.has("Mithalan God beated", player) and
+            state.has("Drunian God beated", player) and
+            state.has("Sun God beated", player) and
+            state.has("The Golem beated", player)
+            )
+
+
+def _has_mini_bosses(state, player: int) -> bool:
+    """`player` in `state` has beated every big bosses"""
+    return (state.has("Nautilus Prime beated", player) and
+            state.has("Blaster Peg Prime beated", player) and
+            state.has("Mergog beated", player) and
+            state.has("Mithalan priests beated", player) and
+            state.has("Octopus Prime beated", player) and
+            state.has("Crabbius Maximus beated", player) and
+            state.has("Mantis Shrimp Prime beated", player) and
+            state.has("King Jellyfish God Prime beated", player)
+            )
```

Comment:
```suggestion
def _has_mini_bosses(state, player: int) -> bool:
    """`player` in `state` has beated every big bosses"""
    return state.has_all({"Nautilus Prime beated", "Blaster Peg Prime beated", "Mergog beated", "Mithalan priests beated", "Octopus Prime beated", "Crabbius Maximus beated", "Mantis Shrimp Prime beated", "King Jellyfish God Prime beated"}, player)
```

And similar for `_has_big_bosses` and `_has_secrets`
(and also add line breaks in there, I'm not on an IDE right now so I couldn't include those properly)